collector/src/main/java/com/bitmovin/analytics/retryBackend/RetryQueue.kt

```diff
@@ -0,0 +1,81 @@
+package com.bitmovin.analytics.retryBackend
+
+import android.util.Log
+import com.bitmovin.analytics.utils.Util
+import java.lang.Exception
+import java.util.Calendar
+import java.util.Date
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.Comparator
+import kotlin.math.pow
+
+class RetryQueue {
+    private val TAG = "RetryQueue"
+    private val lock = ReentrantLock()
+
+    fun getMaxSampleNumber() = Util.MAX_RETRY_SAMPLES
+
+    private var retrySamplesList = mutableListOf<RetrySample<Any>>()
+    private val sampleComparator = Comparator<RetrySample<Any>> { a, b ->
+        when {
+            (a.scheduledTime == b.scheduledTime) -> 0
+            (a.scheduledTime.before(b.scheduledTime)) -> -1
+            else -> 1
+        }
+    }
+
+    fun addSample(retrySample: RetrySample<Any>) {
+
+        try {
+            lock.lock()
+            retrySample.retry++
+
+            val backOffTime = minOf(2.toDouble().pow(retrySample.retry).toInt(), 64)
+            retrySample.totalTime += backOffTime
+            // more than 5min in queue
+            if (retrySample.totalTime < Util.MAX_RETRY_TIME) {
```

Comment:
can we move the constant to this class and not to the `Util` class? Feels weird that we have a configuration constant in a `Util` class

Reply:
moved constants from util https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/132/commits/626828002ef8e7411471c2172d47085a68222dc0
also added new constant `MAX_BACKOFF_INTERVAL = 64` and fixed its value (bacause I'm blind 🙈 )
https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/132/commits/f93573e337d8ec4813d97988c1890eedca2b8b5e